src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -976,8 +976,9 @@ public void Sort(IComparer<T> comparer)
             }
             Contract.EndContractBlock();
 
-            if( _size > 0) {
+            if (_size > 0) {
                 ArraySortHelper<T>.Sort(_items, 0, _size, comparison);
+                _version++;
```

Comment:
OK, should I move version increment out of the `if` block?

And what about my proposal to check that the size is at least 2? Is it superfluous?

    public void Sort(int index, int count, IComparer<T> comparer)
    {
        ...

        if (count > 1)
        {
            Array.Sort<T>(_items, index, count, comparer);
        }
        _version++;
    }

    public void Sort(Comparison<T> comparison)
    {
        ...

        if (_size > 1)
        {
            ArraySortHelper<T>.Sort(_items, 0, _size, comparison);
        }
        version++;
    }